src/kafka-handler.ts

```diff
@@ -44,12 +45,18 @@ class KafkaHandler implements DataBroker {
             this.initKafkaConfiguration(config, callback, "consumer");
           });
           this.producer.on("error", (e) => { 
+            if (isScheduled == true) {
+              console.log("An operation was already scheduled. No need to do it again.");
+              return;
+            }
+            this.producer.close();
+            isScheduled = true;
             console.error("Error: ", e); 
-            console.log("Will try again in a few seconds.");
-            setTimeout(() => {
-              console.log("Trying again.");
-              this.initKafkaConfiguration(config, callback, "producer");
-            }, 1000)
+            console.log("Will try again to create Kafka producer in a few seconds.");
+              setTimeout(() => {
+                console.log("Trying again.");
+                this.initKafkaConfiguration(config, callback, "producer");
+              }, 100);
```

Comment:
The PR message says this timeout was increased, but here it is decreased

Reply:
Oh